### DISCUSSION TITLE
Input Checkbox indeterminate Property

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#881] [Ideas]
Hi

Discovered that checkbox element, have 3 states:
checked, unchecked, or indeterminate

![image](https://user-images.githubusercontent.com/89095737/177655638-9a51da0e-eab6-4a71-aba5-a3b0e086ff89.png)

I thought it was some kind of hack, but I found information [here](https://www.w3schools.com/jsref/prop_checkbox_indeterminate.asp)

Wouldn't it be interesting to have mavo be able to store those 3 states for checkboxes?

I've made a [codepen](https://codepen.io/galinhalx/pen/MWVKGbm) , but I'm able to store 2 states only

Goal would be to be able to store value for [those kind ](https://codepen.io/galinhalx/pen/VwXexLE)of UI

<img width="134" alt="image" src="https://user-images.githubusercontent.com/89095737/177656635-4744b41d-4dd4-42f0-b183-008f81196ec7.png">

**Comment by LeaVerou** [comment]
Indeterminate usually represents something else, e.g. you have a series of other checkboxes, some of which are checked and some of which are not, so the "select all" checkbox would be indeterminate. It's not a state that ever occurs organically through user interaction, it can only be set programmatically. So usually you don't want to save that as data, but the data that caused it to be indeterminate.
For an example of setting indeterminate with Mavo, check out the select all checkbox here: https://test.mavo.io/

**Reply by GalinhaLX** [reply]
Thanks a lot for this explanation @LeaVerou

**Comment by GalinhaLX** [comment]
just because I found it fun to do:

based on @LeaVerou and @DmitrySharabin other codepens, and mix of everything

[sharing my codepen ](https://codepen.io/galinhalx/pen/ZExWWqw) about some 3 states mavo radio buttons (checked, unchecked, or indeterminate), where value is saved with mavo
